docs/AccessEnvironmentVariables.md

```diff
@@ -0,0 +1,174 @@
+# Access environment variables through Central Configuration Inversion
+
+This document explains how to properly access environment variables and manage configuration in the dd-trace-rb library with the central configuration inversion initiative.
+
+## Overview
+
+The dd-trace-rb library implements **central configuration inversion** through the `ConfigHelper` module to ensure all environment variable access is centralized, documented, and validated. This approach prevents direct `ENV` access and enforces that all supported environment variables are properly registered.
+
+Central configuration inversion name means that there is a single, centralized source of truth for configuration, and libraries use that source to accept configurations or not. This inverts the previous process, where configurations were defined in each library and we attempted to create a source of truth from the existing code, often leading to incomplete documentation.
+
+## Key Components
+
+### 1. Environment variables access
+
+Instead of accessing `ENV` directly, use `Datadog.get_environment_variable` to read environment variables:
+
+```ruby
+# ❌ Bad: Direct ENV access
+api_key = ENV['DD_API_KEY']
+timeout = ENV.fetch('DD_TIMEOUT', '30')
+
+# ✅ Good: Using Datadog.get_environment_variable
+api_key = Datadog.get_environment_variable('DD_API_KEY')
+timeout = Datadog.get_environment_variable('DD_TIMEOUT', '30')
```

Comment:
Do you think it would make it easier to read, and easier to find env vars if we try to stick to the built-in `ENV`'s naming?
```suggestion
api_key = Datadog::ENV['DD_API_KEY']
timeout = Datadog::Env.fetch('DD_TIMEOUT', '30')
```
In this suggestion, someone grepping this repo for `ENV` would still find all env var accesses, even if they don't know anything about our new env var access control. Also, it allow us to keep using the familiar syntax of `#[]` and `#fetch`.